packages/aws/src/s3.ts

```diff
@@ -170,15 +170,27 @@ export class S3BlobStore implements BlobStore {
   }
 
   private async move(keys: { from: string; to: string }) {
-    await this.client.copyObject({
-      Bucket: this.bucket,
-      CopySource: `${this.bucket}/${keys.from}`,
-      Key: keys.to,
-    })
-    await this.client.deleteObject({
-      Bucket: this.bucket,
-      Key: keys.from,
-    })
+    try {
+      await this.client.copyObject({
+        Bucket: this.bucket,
+        CopySource: `${this.bucket}/${keys.from}`,
+        Key: keys.to,
+      })
+      await this.client.deleteObject({
+        Bucket: this.bucket,
+        Key: keys.from,
+      })
+    } catch (err) {
+      handleErr(err)
+    }
+  }
+}
+
+const handleErr = (err: unknown) => {
+  if (err && typeof err === 'object' && err['Code'] === 'NoSuchKey') {
```

Comment:
I believe this is totally typesafe:
```suggestion
  if (err?.['Code'] === 'NoSuchKey') {
```

Reply:
typescript yelled at me for that :/